embedded-postgresql/src/test/java/com/playtika/test/postgresql/EmbeddedPostgreSQLBootstrapConfigurationTest.java

```diff
@@ -0,0 +1,127 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static com.playtika.test.postgresql.PostgreSQLProperties.BEAN_NAME_EMBEDDED_POSTGRESQL;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    classes = EmbeddedPostgreSQLBootstrapConfigurationTest.TestConfiguration.class
+    , properties = "spring.profiles.active=enabled"
+)
+public class EmbeddedPostgreSQLBootstrapConfigurationTest {
+
+    @Autowired
+    ConfigurableListableBeanFactory beanFactory;
+
+    @Autowired
+    JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    ConfigurableEnvironment environment;
+
+    @Test
+    public void shouldConnectToPostgreSQL() {
+        assertThat(jdbcTemplate.queryForObject("select version()", String.class)).contains("PostgreSQL");
+    }
+
+    @Test
+    public void shouldSaveAndGetUnicode() {
+        jdbcTemplate.execute("CREATE TABLE employee(id INT, name VARCHAR(64));");
+        jdbcTemplate.execute("insert into employee (id, name) values (1, 'some data \uD83D\uDE22');");
+
+        assertThat(jdbcTemplate.queryForObject("select name from employee where id = 1", String.class)).isEqualTo("some data \uD83D\uDE22");
+    }
+
+    @Test
+    public void propertiesAreAvailable() {
+        assertThat(environment.getProperty("embedded.postgresql.port")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.postgresql.host")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.postgresql.schema")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.postgresql.user")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.postgresql.password")).isNotEmpty();
+    }
+
+    @Test
+    public void shouldSetupDependsOnForAllDataSources() throws Exception {
+        String[] beanNamesForType = beanFactory.getBeanNamesForType(DataSource.class);
+        assertThat(beanNamesForType)
+            .as("Custom datasource should be present")
+            .hasSize(1)
+            .contains("customDatasource");
+        asList(beanNamesForType).forEach(this::hasDependsOn);
+    }
+
+    private void hasDependsOn(String beanName) {
+        assertThat(beanFactory.getBeanDefinition(beanName).getDependsOn())
+            .isNotNull()
+            .isNotEmpty()
+            .contains(BEAN_NAME_EMBEDDED_POSTGRESQL);
+    }
+
+    @EnableAutoConfiguration
```

Comment:
`@EnableAutoConfiguration` should be enabled by default in `@SpringBootTest`

Reply:
It is not